### PR TITLE
refactoring to range loop

### PR DESCRIPTION
**Pull request by Joilnen**

### CODE REVIEW

File: OgreMain/src/OgreCompositorInstance.cpp [hunk 1]
```diff
@@ -906,11 +899,10 @@ void CompositorInstance::freeResources(bool forResizeOnly, bool clearReserveText
     // required (saves some time & memory thrashing / fragmentation on resize)
 
     const CompositionTechnique::TextureDefinitions& tdefs = mTechnique->getTextureDefinitions();
-    CompositionTechnique::TextureDefinitions::const_iterator it = tdefs.begin();
-    for (; it != tdefs.end(); ++it)
+    // CompositionTechnique::TextureDefinitions::const_iterator it = tdefs.begin();
+    // for (; it != tdefs.end(); ++it)
```
Comment (akacoding) [inline OgreMain/src/OgreCompositorInstance.cpp:903]: You forgot to remove the old code as comment here.

File: OgreMain/src/OgreCompositorInstance.cpp [hunk 2]
```diff
@@ -1194,30 +1183,30 @@ void CompositorInstance::removeListener(Listener *l)
 //-----------------------------------------------------------------------
 void CompositorInstance::_fireNotifyMaterialSetup(uint32 pass_id, MaterialPtr &mat)
 {
-    Listeners::iterator i, iend=mListeners.end();
-    for(i=mListeners.begin(); i!=iend; ++i)
-        (*i)->notifyMaterialSetup(pass_id, mat);
+    // Listeners::iterator i, iend=mListeners.end();
```
Comment (akacoding) [inline OgreMain/src/OgreCompositorInstance.cpp:1186]: You forgot to remove the old code as comment here.

File: OgreMain/src/OgreCompositorInstance.cpp [hunk 3]
```diff
@@ -1194,30 +1183,30 @@ void CompositorInstance::removeListener(Listener *l)
...
+    for(auto *l : mListeners)
+        l->notifyMaterialSetup(pass_id, mat);
 }
 //-----------------------------------------------------------------------
 void CompositorInstance::_fireNotifyMaterialRender(uint32 pass_id, MaterialPtr &mat)
 {
-    Listeners::iterator i, iend=mListeners.end();
-    for(i=mListeners.begin(); i!=iend; ++i)
-        (*i)->notifyMaterialRender(pass_id, mat);
+    // Listeners::iterator i, iend=mListeners.end();
```
Comment (akacoding) [inline OgreMain/src/OgreCompositorInstance.cpp:1193]: You forgot to remove the old code as comment here.

File: OgreMain/src/OgreCompositorInstance.cpp [hunk 4]
```diff
@@ -1194,30 +1183,30 @@ void CompositorInstance::removeListener(Listener *l)
...
+    for(auto *l : mListeners)
+        l->notifyMaterialRender(pass_id, mat);
 }
 //-----------------------------------------------------------------------
 void CompositorInstance::_fireNotifyResourcesCreated(bool forResizeOnly)
 {
-    Listeners::iterator i, iend=mListeners.end();
-    for(i=mListeners.begin(); i!=iend; ++i)
-        (*i)->notifyResourcesCreated(forResizeOnly);
+    // Listeners::iterator i, iend=mListeners.end();
```
Comment (akacoding) [inline OgreMain/src/OgreCompositorInstance.cpp:1200]: You forgot to remove the old code as comment here.

File: OgreMain/src/OgreCompositorInstance.cpp [hunk 5]
```diff
@@ -1194,30 +1183,30 @@ void CompositorInstance::removeListener(Listener *l)
...
+    for(auto *l : mListeners)
+        l->notifyResourcesCreated(forResizeOnly);
 }
 //-----------------------------------------------------------------------
 void CompositorInstance::_fireNotifyResourcesReleased(bool forResizeOnly)
 {
-    Listeners::iterator i, iend=mListeners.end();
-    for(i=mListeners.begin(); i!=iend; ++i)
-        (*i)->notifyResourcesReleased(forResizeOnly);
+    // Listeners::iterator i, iend=mListeners.end();
```
Comment (akacoding) [inline OgreMain/src/OgreCompositorInstance.cpp:1207]: You forgot to remove the old code as comment here.

File: OgreMain/src/OgreCompositorInstance.cpp [hunk 6]
```diff
@@ -392,13 +391,11 @@ void CompositorInstance::collectPasses(TargetOperation &finalState, const Compos
             /// Create local material
             MaterialPtr localMat = createLocalMaterial(srcmat->getName());
             /// Copy and adapt passes from source material
-            Technique::Passes::const_iterator i;
-            for(i = srctech->getPasses().begin(); i != srctech->getPasses().end(); ++i)
+            for(auto *scrpass : srctech->getPasses())
```
[the remark this answers is not in the document]
Comment (Joilnen) [reply OgreMain/src/OgreCompositorInstance.cpp]: changed

**Pull request merged**